esp-hal/src/dma/mod.rs

```diff
@@ -1860,22 +1860,197 @@ pub trait DmaRxBuffer {
 }
 
 /// Error returned from Dma[Rx|Tx|RxTx]Buf operations.
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub enum DmaBufError {
     /// More descriptors are needed for the buffer size
     InsufficientDescriptors,
     /// Descriptors or buffers are not located in a supported memory region
     UnsupportedMemoryRegion,
 }
 
+#[derive(Debug)]
+struct DescriptorSet<'a> {
```

Comment:
This struct should probably just be a list of free functions, maybe in a private `util` module.

Having this struct take and hold an exclusive reference adds an extra layer of code that needs modifying when adding a new feature

Reply:
Not sure I agree here.

Reply:
Ah, free functions would also benefit the `DescriptorChain` struct.